src/components/recovery/RecoveryContext/useRecoveryState.ts

```diff
@@ -35,34 +33,10 @@ export function useRecoveryState(delayModifiers?: Array<Delay>): {
     setRefetchDep((prev) => !prev)
   }, [])
 
-  const data = useAsync<RecoveryState>(
-    () => {
-      if (!delayModifiers || delayModifiers.length === 0 || !chain?.transactionService || !web3ReadOnly) {
-        return
-      }
-
-      return getRecoveryState({
-        delayModifiers,
-        transactionService: chain.transactionService,
-        safeAddress,
-        provider: web3ReadOnly,
-        chainId: safe.chainId,
-        version: safe.version,
-      })
-    },
-    // eslint-disable-next-line react-hooks/exhaustive-deps
-    [
-      delayModifiers,
-      counter,
-      refetchDep,
-      chain?.transactionService,
-      web3ReadOnly,
-      safeAddress,
-      safe.chainId,
-      safe.version,
-    ],
-    false,
-  )
+  // Reload recovery data when a Guardian transaction occurs
+  useEffect(() => {
+    return recoverySubscribe(RecoveryEvent.PROCESSED, refetch)
+  }, [refetch])
```

Comment:
With this in place, do we still need the tx history listener?

Reply:
Yes, the history listener is required for cancellations as we do not know what kind of transaction is executed.